### PR TITLE
[2296] Allow users of the code editor to save and continue working

### PR DESCRIPTION
**Pull request by aachavarria**
https://github.com/craftercms/craftercms/issues/2296
https://github.com/craftercms/craftercms/issues/4306

### CODE REVIEW

File: static-assets/components/cstudio-forms/controls/rte-tinymce5.js [hunk 1]
```diff
@@ -701,7 +701,6 @@ CStudioAuthoring.Module.requireModule(
         _onChangeVal: function(evt, obj) {
           obj.edited = true;
           this._onChange(evt, obj);
-          obj.owner.pendingChanges();
```
Comment (rart) [inline static-assets/components/cstudio-forms/controls/rte-tinymce5.js:716]: ?

File: static-assets/components/cstudio-forms/forms-engine.js [hunk 2]
```diff
@@ -544,13 +544,6 @@ var CStudioForms =
             YAHOO.util.Dom.addClass(indicatorEl, 'fa-check');
           }
         }
-      },
-
-      pendingChanges: function() {
-        if (this.iceWindowCallback && this.iceWindowCallback.pendingChanges) {
-          let callback = getCustomCallback(this.iceWindowCallback.pendingChanges);
-          callback();
-        }
```
Comment (rart) [inline static-assets/components/cstudio-forms/forms-engine.js:553]: ?

File: static-assets/components/cstudio-forms/forms-engine.js [hunk 3]
```diff
@@ -1164,14 +1157,6 @@ var CStudioForms =
         form.definition.pageLocation = this._getPageLocation(path);
         form.containerEl = document.getElementById('formContainer');
 
-        $(form.containerEl).on('change', '.datum', () => {
-          var flag = isModified();
-          if (flag && iceWindowCallback && iceWindowCallback.pendingChanges) {
-            let callback = getCustomCallback(iceWindowCallback.pendingChanges);
-            callback();
-          }
-        });
```
Comment (rart) [inline static-assets/components/cstudio-forms/forms-engine.js:1173]: ?

**Review by rart — COMMENTED** [review]
Please speak to what are all the deletions on the form-engine related files.

**Review by rart — APPROVED** [review]

**Comment by sumerjabri** [comment]
@aachavarria please fix the conflict.

**Comment by aachavarria** [comment]
@sumerjabri done.

**Pull request merged**